sentry-android-okhttp/src/main/java/io/sentry/android/okhttp/SentryOkHttpEventListener.kt

```diff
@@ -0,0 +1,370 @@
+package io.sentry.android.okhttp
+
+import io.sentry.HubAdapter
+import io.sentry.IHub
+import io.sentry.SpanStatus
+import okhttp3.Call
+import okhttp3.Connection
+import okhttp3.EventListener
+import okhttp3.Handshake
+import okhttp3.HttpUrl
+import okhttp3.Protocol
+import okhttp3.Request
+import okhttp3.Response
+import java.io.IOException
+import java.net.InetAddress
+import java.net.InetSocketAddress
+import java.net.Proxy
+
+/**
+ *  Logs network performance event metrics to Sentry
+ *
+ *  Usage - add instance of [SentryOkHttpEventListener] in [OkHttpClient.eventListener]
+ *
+ * ```
+ * val client = OkHttpClient.Builder()
+ *     .eventListener(SentryOkHttpEventListener())
+ *     .addInterceptor(SentryOkHttpInterceptor())
+ *     .build()
+ * ```
+ *
+ * If you already use a [OkHttpClient.eventListener], you can pass it in the constructor.
+ *
+ * ```
+ * val client = OkHttpClient.Builder()
+ *     .eventListener(SentryOkHttpEventListener(myEventListener))
+ *     .addInterceptor(SentryOkHttpInterceptor())
+ *     .build()
+ * ```
+ */
+@Suppress("TooManyFunctions")
+class SentryOkHttpEventListener(
+    private val hub: IHub = HubAdapter.getInstance(),
+    private val originalEventListenerCreator: ((call: Call) -> EventListener)? = null
+) : EventListener() {
+
+    private var originalEventListener: EventListener? = null
+
+    companion object {
+        internal const val PROXY_SELECT_EVENT = "proxySelect"
+        internal const val DNS_EVENT = "dns"
+        internal const val SECURE_CONNECT_EVENT = "secureConnect"
+        internal const val CONNECT_EVENT = "connect"
+        internal const val CONNECTION_EVENT = "connection"
+        internal const val REQUEST_HEADERS_EVENT = "requestHeaders"
+        internal const val REQUEST_BODY_EVENT = "requestBody"
+        internal const val RESPONSE_HEADERS_EVENT = "responseHeaders"
+        internal const val RESPONSE_BODY_EVENT = "responseBody"
+
+        internal val eventMap: MutableMap<Call, SentryOkHttpEvent> = HashMap()
```

Comment:
```suggestion
        internal val eventMap: MutableMap<Call, SentryOkHttpEvent> = ConcurrentHashMap()
```
Do we need thread safe?